extensions/histogram/src/main/java/io/druid/query/aggregation/histogram/ApproximateHistogramAggregatorFactory.java

```diff
@@ -115,6 +116,36 @@ public AggregatorFactory getCombiningFactory()
   }
 
   @Override
+  public AggregatorFactory getMergingFactory(AggregatorFactory other) throws AggregatorFactoryNotMergeableException
+  {
+    if (other.getName().equals(this.getName()) && other instanceof ApproximateHistogramAggregatorFactory) {
+      ApproximateHistogramAggregatorFactory castedOther = (ApproximateHistogramAggregatorFactory) other;
+
+      if (Float.compare(castedOther.lowerLimit, lowerLimit) == 0 &&
+          numBuckets != castedOther.numBuckets &&
+          resolution != castedOther.resolution &&
```

Comment:
I'm trying to match everything due to my ignorance of the histogram algorithm implementation used. Then, it appears, I can change it to 
```
  public AggregatorFactory getMergingFactory(AggregatorFactory other) throws AggregatorFactoryNotMergeableException
  {
    if (other.getName().equals(this.getName()) && other instanceof ApproximateHistogramAggregatorFactory) {
      ApproximateHistogramAggregatorFactory castedOther = (ApproximateHistogramAggregatorFactory) other;

      return new ApproximateHistogramFoldingAggregatorFactory(
          name,
          name,
          Math.max(resolution, castedOther.resolution),
          numBuckets,
          Math.min(lowerLimit, castedOther.lowerLimit),
          Math.max(upperLimit, castedOther.upperLimit)
      );

    } else {
      throw new AggregatorFactoryNotMergeableException(
          "can't merge [%s : %s] and [%s : %s]",
          this.getName(),
          this.getClass().getName(),
          other.getName(),
          other.getClass().getName()
      );
    }
  }
```

does that look OK?